src/rosdiscover/recover/loader.py

```diff
@@ -114,7 +114,7 @@ def _load_variable_reference(self, dict_: t.Mapping[str, t.Any]) -> SymbolicVari
         assert dict_["kind"] == "variable-reference"
         type_ = SymbolicValueType.from_name(dict_["type"])
         return SymbolicVariableReference(
-            variable=dict_["variable"],
+            variable=dict_["qualified_name"],
```

Comment:
we probably don't want to change anything now, but we should be careful about mixing `_` and `-` as word separators in the JSON output.